CHANGELOG.md

```diff
@@ -8,6 +8,7 @@ Mapbox welcomes participation and contributions from everyone.
 
 #### Bug fixes and improvements
 - :warning: Realigned all `NavigationCamera` transitions to not fallback to `Idle` if a transition is canceled to keep the behavior consistent and predictable (state transitions used to be cancelable while frame transitions weren't). If you need the `NavigationCamera` to reset to `Idle` on external interactions or cancellations, use `NavigationBasicGesturesHandler` or `NavigationScaleGestureHandler` [#5607](https://github.com/mapbox/mapbox-navigation-android/pull/5607)
+- Improved Reroute immediately switch to alternative when user turns to it. [#5645](https://github.com/mapbox/mapbox-navigation-android/pull/5645)
```

Comment:
Maybe highlight the fact that it works only for default reroute controller? 
```suggestion
- Improved reroute for a default controller. `MapboxRerouteController.kt`  immediately switches to an alternative route when a user turns to it. Now the SDK provides enough data to implement the same feature in a custom reroute controller. [#5645](https://github.com/mapbox/mapbox-navigation-android/pull/5645)
```

Reply:
The suggesting makes sense only if https://github.com/mapbox/mapbox-navigation-android/pull/5645#discussion_r841507041 accepted

Reply:
> Maybe highlight the fact that it works only for default reroute controller?

make sense 👍

Reply:
```suggestion
- Improved reroute experience for a default controller. `MapboxRerouteController` now immediately switches to an alternative route when a user turns to it, without making an unnecessary route request. [#5645](https://github.com/mapbox/mapbox-navigation-android/pull/5645)
```

Refactoring @VysotskiVadim's suggestion, that last part wasn't actually true yet since the IDs are hidden.